tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedEventMatcher.cs

```diff
@@ -0,0 +1,167 @@
+                            {
+                                case "commandName":
+                                    commandFailedEvent.CommandName.Should().Be(element.Value.AsString);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unexpected commandStartedEvent field: '{element.Name}'");
+                            }
+                        }
+                        break;
+                    default:
+                        throw new FormatException($"Unrecognized event type: '{actualEvent.GetType()}'");
+                }
+            }
+        }
+
+        private string GetAssertionErrorMessage(List<object> actualEvents, BsonArray expectedEventsDocuments)
+        {
+            var jsonWriterSettings = new JsonWriterSettings { Indent = true };
+
+            var actualEventsDocuments = new BsonArray();
+            foreach (var actualEvent in actualEvents)
+            {
+                switch (actualEvent)
+                {
+                    case CommandStartedEvent commandStartedEvent:
+                        var commandStartedDocument = new BsonDocument
+                        {
+                            { "command", commandStartedEvent.Command },
+                            { "commandName", commandStartedEvent.CommandName },
+                            { "databaseName", commandStartedEvent.DatabaseNamespace.DatabaseName }
+                        };
+                        actualEventsDocuments.Add(new BsonDocument("commandStartedEvent", commandStartedDocument));
+                        break;
+                    case CommandSucceededEvent commandSucceededEvent:
+                        var commandSucceededDocument = new BsonDocument
+                        {
+                            { "reply", commandSucceededEvent.Reply },
+                            { "commandName", commandSucceededEvent.CommandName }
+                        };
+                        actualEventsDocuments.Add(new BsonDocument("commandSucceededEvent", commandSucceededDocument));
+                        break;
+                    case CommandFailedEvent commandFailedEvent:
+                        var commandFailedDocument = new BsonDocument
+                        {
+                            { "commandName", commandFailedEvent.CommandName }
+                        };
+                        actualEventsDocuments.Add(new BsonDocument("commandFailedEvent", commandFailedDocument));
+                        break;
+                    default:
+                        throw new FormatException($"Unrecognized event type: '{actualEvent.GetType()}'");
+                }
+            }
+
+            return $"{Environment.NewLine}Expected events to be: {expectedEventsDocuments.ToJson(jsonWriterSettings)}" +
+                   $"{Environment.NewLine}But found: {actualEventsDocuments.ToJson(jsonWriterSettings)}";
```

Comment:
`NewLine` at the **beginning** of the line doesn't make sense to me. Shouldn't this be:

    return
        $"Expected events to be: {expectedEventsDocuments.ToJson(jsonWriterSettings)}{Environment.NewLine}" +
        $"But found: {actualEventsDocuments.ToJson(jsonWriterSettings)}{Environment.NewLine}";